L80-csharp-load-balancing.md

```diff
@@ -0,0 +1,259 @@
+    }
+    public class PickFirstBalancer : LoadBalancer {
+        public PickFirstBalancer(IConnectionController channel, ILoggerFactory loggerFactory);
+        public override void ResolverError(Exception exception);
+        public override void UpdateChannelState(ChannelState state);
+        public override void UpdateSubChannelState(SubChannel subChannel, SubChannelState state);
+    }
+    public class PickFirstBalancerFactory : LoadBalancerFactory {
+        public PickFirstBalancerFactory(ILoggerFactory loggerFactory);
+        public override string Name { get; }
+        public override LoadBalancer Create(IConnectionController controller, IDictionary<string, object> options);
+    }
+    public class PickResult {
+        public PickResult(SubChannel? subChannel, DnsEndPoint? endPoint, Action<CompleteContext>? onComplete);
+        public DnsEndPoint? EndPoint { get; }
+        public SubChannel? SubChannel { get; }
+        public void Complete(CompleteContext context);
+    }
+    public class RoundRobinBalancer : SubChannelsLoadBalancer {
+        public RoundRobinBalancer(IConnectionController channel, ILoggerFactory loggerFactory);
+        protected override SubChannelPicker CreatePicker(List<(DnsEndPoint Address, SubChannel SubChannel)> readySubChannels);
+    }
+    public class RoundRobinBalancerFactory : LoadBalancerFactory {
+        public RoundRobinBalancerFactory(ILoggerFactory loggerFactory);
+        public override string Name { get; }
+        public override LoadBalancer Create(IConnectionController controller, IDictionary<string, object> options);
+    }
+    public class StaticAddressResolver : AddressResolver {
+        public StaticAddressResolver(IEnumerable<DnsEndPoint> addresses);
+        public override Task RefreshAsync(CancellationToken cancellationToken);
+        public override IDisposable Subscribe(IObserver<AddressResolverResult> observer);
+        protected override void Dispose(bool disposing);
+    }
+    public class StaticAddressResolverFactory : AddressResolverFactory {
+        public StaticAddressResolverFactory(IEnumerable<DnsEndPoint> addresses);
+        public override string Name { get; }
+        public override AddressResolver Create(Uri address, AddressResolverOptions options);
+    }
+    public class SubChannel : IDisposable {
+        public BalancerAttributes Attributes { get; }
+        public DnsEndPoint? CurrentEndPoint { get; }
+        public ConnectivityState State { get; }
+        public Task ConnectAsync(CancellationToken cancellationToken);
+        public void Dispose();
+        public IList<DnsEndPoint> GetAddresses();
+        public void UpdateAddresses(IReadOnlyList<DnsEndPoint> addresses);
+        public override string ToString();
+    }
+    public abstract class SubChannelPicker {
+        protected SubChannelPicker();
+        public abstract PickResult Pick(PickContext context);
+    }
+    public abstract class SubChannelsLoadBalancer : LoadBalancer {
```

Comment:
What is this for?

Reply:
It is like the base balancer in go - https://github.com/grpc/grpc-go/blob/3dd75a6888ce5d1b5195c5cf72241d9e36f68e42/balancer/base/balancer.go

It's a base type for balancers that have a channel per address and have a policy for picking from them.

Reply:
Okay.  Some comments might help here.